src/main/java/com/atlassian/db/replica/api/aurora/AuroraReplicasDiscoverer.java

```diff
@@ -0,0 +1,55 @@
+package com.atlassian.db.replica.api.aurora;
+
+import com.atlassian.db.replica.internal.aurora.AuroraEndpoint;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.atlassian.db.replica.internal.aurora.AuroraEndpoints.instanceEndpoint;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * It allows to pull Aurora Replicas cluster information
+ */
+public final class AuroraReplicasDiscoverer {
+    private final Connection connection;
+
+    public AuroraReplicasDiscoverer(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * It provides list of replica endpoints based on reader endpoint and server ids
+     * @param readerEndpoint - Aurora reader endpoint
+     * @return list of replicas endpoints
+     * @throws SQLException
+     */
+    public List<AuroraEndpoint> fetchReplicasEndpoints(String readerEndpoint) throws SQLException {
```

Comment:
`AuroraReplicasDiscoverer` is public API so its public signatures are API too. This includes the `AuroraEndpoint` class.

Reply:
AuroraEndpoint -> String
or expose additional class 🤔 I don't mind returning string here, to be honest.